client/web/src/util/permission.ts

```diff
@@ -0,0 +1,15 @@
+import { findIndex } from 'lodash'
+
+import type { AuthenticatedUser } from '../auth'
+import type { RbacPermission } from '../rbac/constants'
+
+export const doesUserHavePermission = (
+    user: Pick<AuthenticatedUser, 'permissions'> | null,
+    permissionToCheckFor: RbacPermission
+): boolean => {
+    if (user === null) {
+        return false
+    }
+
+    return findIndex(user.permissions.nodes, permission => permission.displayName === permissionToCheckFor) !== -1
```

Comment:
Why lodash here? Couldn't we just use builtins?

```suggestion
    return user.permissions.nodes.some(permission => permission.displayName === permissionToCheckFor)
```